llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1383,6 +1385,16 @@ static void transformRecipestoEVLRecipes(VPlan &Plan, VPValue &EVL) {
                 VPValue *NewMask = GetNewMask(Red->getCondOp());
                 return new VPReductionEVLRecipe(*Red, EVL, NewMask);
               })
+              .Case<VPInstruction>([&](VPInstruction *VPI) -> VPRecipeBase * {
+                VPValue *LHS, *RHS;
+                if (!match(VPI, m_Select(m_Specific(HeaderMask), m_VPValue(LHS),
+                                         m_VPValue(RHS))))
+                  return nullptr;
+                VPValue *Cond = Plan.getOrAddLiveIn(ConstantInt::getTrue(Ctx));
```

Comment:
Could you add a comment why true is used here? IIUC it is because it is limited to selects on the header mask for now?

Reply:
Your understanding is absolutely correct. :)
And I can't find any other place where vp.merge is used.
7bfd0b20050a9d3859a2065c1b5622986d91f9c1